subprojects/execution/src/main/java/org/gradle/internal/execution/steps/BuildCacheStep.java

```diff
@@ -124,13 +125,25 @@ public Duration getDuration() {
                 })
                 .orElseGet(() -> executeAndStoreInCache(cacheableWork, cacheKey, context))
             )
-            .getOrMapFailure(loadFailure -> {
-                throw new RuntimeException(
-                    String.format("Failed to load cache entry for %s",
-                        work.getDisplayName()),
-                    loadFailure
-                );
-            });
+            .getOrMapFailure(loadFailure -> mapAndThrowFailure(work, cacheKey, loadFailure));
+    }
+
+    private AfterExecutionResult mapAndThrowFailure(UnitOfWork work, BuildCacheKey cacheKey, Throwable loadFailure) {
+        String cacheName = "build cache";
+        String loadFailureMessage = loadFailure.getMessage();
+        if (loadFailure instanceof BuildCacheLoadException) {
+            cacheName = ((BuildCacheLoadException) loadFailure).getCacheName();
+            loadFailureMessage = loadFailure.getCause().getMessage();
```

Comment:
Why dp we make up a complicated message in `BuildCacheLoadException` if we do not use it?

What is wrong with always throwing a `RuntimeException` here with what information we add on this level (= work item, cache key) and letting `BuildCacheLoadException` carry the rest of the context (=which cache)?

Reply:
The message from `BuildCacheLoadException` is seen if you use `--stacktrace`.

The problem is, that if you don't use `--stacktrace` flag, you don't see enough info as reported in the isseue.

Reply:
I simplified this a bit and removed `BuildCacheLoadException`.